compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -368,7 +368,10 @@ impl<'tcx> UnsafetyChecker<'_, 'tcx> {
         if !callee_features.iter().all(|feature| self_features.contains(feature)) {
             self.require_unsafe(
                 UnsafetyViolationKind::General,
-                UnsafetyViolationDetails::CallToFunctionWith,
+                UnsafetyViolationDetails::CallToFunctionWith {
+                    func_did: func_did,
+                    body_did_def_id: self.body_did.to_def_id(),
```

Comment:
isn't this `body_did_def_id` always the `DefId` passed to `check_unsafety`? If so, remove the field here, and pass the value directly to `help_missing_feature` and `note_missing_feature`